pex/resolve/lock_resolver.py

```diff
@@ -258,24 +269,25 @@ def resolve_from_lock(
     # errors under the right interleaving of processes and threads and download artifact targets.
     file_lock_style = FileLockStyle.BSD
 
-    # We eagerly initialize a Pip tool for reasons alluded to above, creation of the Pip tool venv
```

Comment:
I don't quite grok why this is no longer required as a result of this change. Perhaps it wasn't required even before this change? AFAICT the pip venv now gets created lazily in the ArtifactDownloader.pip factory, as it would have before?

Reply:
Exactly, it was not required before. I was attempting to substitute clarity when what i really want is some more sane way to deal with this. The only requirement is that the Pip venv is built before the thread pool is used below and that was satisfied either by passing in a pre-built Pip in the obvious way here or via the default / factory used in `ArtifactDownloader`. What this is all hammering on about is that `ArtifactDownloader` used to just use `get_pip()` lazily in-line where it needed it; that was the real problem. The deeper problem is POSIX locks vs BSD. I could switch AtomicDirectory to just always use BSD and then the issue goes away, but as things stand I try to maintain compatibility with pre-existing users of PEX + NFS in all codepaths possible.